File: backend/src/Microservices/_Common/Common/Entity/Package.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TypeGen.Core.TypeAnnotations;

namespace Common.Entity
{
    [ExportTsInterface(OutputDir = "../../../../../frontend/web/parceldelivery-admin-app/src/app/_models")]
    public class Package
    {
        public string Id { get; set; }
        public string UserId { get; set; }
        public double SizeX { get; set; }
        public double SizeY { get; set; }
        public double SizeZ { get; set; }
        public double Weight { get; set; }
        public bool IsFragile { get; set; }

        public string ShippingRequestId { get; set; }
        public ShippingRequest ShippingRequest { get; set; }
    }
}

File: frontend/mobil/parceldelivery_mobile/lib/models/package.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'package.freezed.dart';
part 'package.g.dart';

@freezed
class Package with _$Package {
  const factory Package({
    required String id,
    required String userId,
    required int sizeX,
    required int sizeY,
    required int sizeZ,
    required int weight,
    required bool isFragile,
    required String shippingRequestId,
  }) = _Package;

  factory Package.fromJson(Map<String, Object?> json) =>
      _$PackageFromJson(json);
}
